### ISSUE TITLE
Mouse scrolling is broken

### ISSUE THREAD

**Issue by melroy89** [telegramdesktop/tdesktop#17254]
### Steps to reproduce

1. Open any chat
2. Try to use your scroll-wheel, to scroll up or down

### Expected behaviour

Scrolling within the message window.

### Actual behaviour

Telegram desktop doesn't respond to any scroll wheel action.

### Operating system

Linux Mint 20.2 (Ubuntu based)

### Version of Telegram Desktop

3.2.4

### Installation source

Static binary from official website

**Comment by ilya-fedin** [comment]
> * Open any chat
> 
> * Try to use your scroll-wheel, to scroll up or down

Works like a charm. Looks like these steps aren't right.

**Comment by danger89** [comment]
Hmm.. restarting Telegram fixed the problem. I could still type messages. 

Is there some kind of crash report or logging file I could refer to?

**Comment by danger89** [comment]
The log file gets overridden.. Maybe you should store at least the current log AND ALSO the previous log file.

**Comment by ilya-fedin** [comment]
Nothing regarding scrolling is logging, it's something handled at toolkit level, not at app level

**Comment by jprando** [comment]
happened here with me too.

what I noticed is that the problem is intermittent, when the scroll does not work, I noticed that the mouse icon is different when it is over the application, and when the mouse icon is not different (from other apps) the scroll works in desktop telegram

I'm using the binary version available here on github in release v3.2.4
Pop!_OS 21.04 x64
Gnome 3.38.5
X11

**Comment by ilya-fedin** [comment]
Without a reliable way to reproduce, it's unlikely to be fixed

**Comment by jprando** [comment]
> Without a reliable way to reproduce, it's unlikely to be fixed

yes, i'll keep an eye out here and if i can find a way to reproduce it, i'll share it here 😉

**Comment by danger89** [comment]
@ilya-fedin yes, you could also think along how to debug this issue better.

**Comment by ilya-fedin** [comment]
@danger89 I don't imagine how this could happen and how to debug this, tdesktop just reacts to mouse events sent by Qt. I have a feel this is a Qt bug.

**Comment by AndreyKopeyko** [comment]
I have the same problem running 3.2.2 and 3.2.5 versions.

**Comment by AndreyKopeyko** [comment]
> @danger89 I don't imagine how this could happen and how to debug this, tdesktop just reacts to mouse events sent by Qt. I have a feel this is a Qt bug.

It should happened in 3.1.8 version also if it was a Qt bag. But 3.1.8 scrolls nicely.

**Comment by ilya-fedin** [comment]
Qt was updated in 3.2.0

**Comment by ilya-fedin** [comment]
> libqt5gui5:amd64 5.12.8+dfsg-0ubuntu1

tdesktop doesn't use your system Qt, it has Qt 6.2 embedded in the binary

**Comment by ilya-fedin** [comment]
>     2\. put your note asleep - just close the lid

Can you be more specific - you mean suspend or hibernation? I don't have any action set to the lid.

**Comment by AndreyKopeyko** [comment]
> > 1. put your note asleep - just close the lid
> 
> Can you be more specific - you mean suspend or hibernation? I don't have any action assigned to the lid.

suspend. No hibernation.

**Comment by ilya-fedin** [comment]
> suspend. No hibernation.

Sorry, but I can't reproduce :(
Works like a charm after suspend on my hardware.

**Comment by danger89** [comment]
> Qt was updated in 3.2.0

This could be why.. 

Maybe build a debug app (if possible, so we can catch any eerros)? Or did you see anything strange in the telegram desktop log.txt?

**Comment by AndreyKopeyko** [comment]
> > suspend. No hibernation.
> 
> Sorry, but I can't reproduce :( Works like a charm after suspend on my hardware.

Sorry indeed.
How can I get additional info for you? Does desktop write some logs I can send you back?

**Comment by ilya-fedin** [comment]
> Maybe build a debug app (if possible, so we can catch any eerros)? Or did you see anything strange in the telegram desktop log.txt?

I don't imagine how a debug build would help, this is not a crash, so debug symbols won't help. log.txt won't help either since, as I said previously, nothing regarding scrolling is logged.

> How can I get additional info for you?

I don't know :(. If this is a Qt bug, only Qt developers know how to debug this...

**Comment by danger89** [comment]
yea I could find some debugging techniques within the Qt docs:
https://doc.qt.io/qt-5/debug.html

Just maybe `QT_DEBUG_PLUGINS=1` environment variable could help... Maybe..

**Comment by ilya-fedin** [comment]
All QT_DEBUG_PLUGINS=1 is outputs debug info about plugins loading

**Comment by ilya-fedin** [comment]
maybe `QT_LOGGING_RULES='*.debug=true'` would help, I'm not sure...

**Comment by AndreyKopeyko** [comment]
> > suspend. No hibernation.
> 
> Sorry, but I can't reproduce :( Works like a charm after suspend on my hardware.

It solidly happens if notebook was suspended for about 1 hour - the way from home to office.

**Comment by AndreyKopeyko** [comment]
> QT_LOGGING_RULES='*.debug=true'

It produces a lot of output. Hope my SSD is enough to accommodate all of them until scroll freezes...

**Comment by AndreyKopeyko** [comment]
> maybe `QT_LOGGING_RULES='*.debug=true'` would help, I'm not sure...

Pls take a look in the log - https://dev.kopeyko.ru/tmp/telegram-v3.2.5.debug20211119.output

I started telegram as

    kaa@MRGnb2:~/tmp$ QT_LOGGING_RULES='*.debug=true' telegram 2>&1 |tee telegram-v3.2.5.debug20211119.output

All day it work great, but finally wheel scrolling got frozen - after note spend about 45 minutes in suspend mode (the way from office to home) :-( 

Hope it'll help you to investigate the problem.

**Comment by ilya-fedin** [comment]
> Pls take a look in the log - https://dev.kopeyko.ru/tmp/telegram-v3.2.5.debug20211119.output

I see input device re-detection at line 87810 and no scroll wheel events logged since then. I don't know when it happened though since the log has no timestamps.

**Comment by AndreyKopeyko** [comment]
> > Pls take a look in the log - https://dev.kopeyko.ru/tmp/telegram-v3.2.5.debug20211119.output
> 
> I see input device re-detection at line 87810 and no scroll wheel events logged since then. I don't know when it happened though since the log has no timestamps.

How I can enable timestamps?

**Comment by ilya-fedin** [comment]
looks like `QT_MESSAGE_PATTERN="[%{time}] %{if-category}%{category}: %{endif}%{message}"`

**Comment by AndreyKopeyko** [comment]
> looks like `QT_MESSAGE_PATTERN="[%{time}] %{if-category}%{category}: %{endif}%{message}"`

Right, this is better. You'll get it.

**Comment by thenetengineer** [comment]
Same issue on Ubuntu 20.04.3 LTS - in this case it's a workstation that is on 24/7, I have rebooted, updated and still have the same issue.

**Comment by ilya-fedin** [comment]
> OK, tomorrow morning you'll have the log with a timestamps and with a scroll freeze, both of them.

I hope you understand even if I would have it, I won't be able to fix this

**Comment by AndreyKopeyko** [comment]
Voila! 7 minutes of sleep (in suspend) was enough for wheel scroll to freeze:

    kaa@MRGnb2:~/tmp$ grep -B2 '19T01:53' telegram-v3.2.5.debug$(date +%Y%m%d).output |head -5
    [2021-11-19T01:46:19] qt.qpa.events: Event | XCB_PROPERTY_NOTIFY(28) | sequence: 4726
    [2021-11-19T01:46:19] qt.qpa.events: Event | XCB_PROPERTY_NOTIFY(28) | sequence: 4726
    [2021-11-19T01:53:13] qt.text.font.match: Cache hit level 1
    [2021-11-19T01:53:13] qt.text.font.match: Cache hit level 1
    [2021-11-19T01:53:13] qt.qpa.events: Event | XInput Event(XCB_INPUT_PROPERTY) | sequence: 7837
    kaa@MRGnb2:~/tmp$ 

See full log at https://dev.kopeyko.ru/tmp/telegram-v3.2.5.debug20211119.output2

**Comment by ilya-fedin** [comment]
@AndreyKopeyko the new log doesn't have any scroll events, at all

**Comment by AndreyKopeyko** [comment]
> @AndreyKopeyko the new log doesn't have any scroll events, at all

Interesting... The command I run was

    QT_LOGGING_RULES='*.debug=true' QT_MESSAGE_PATTERN="[%{time}] %{if-category}%{category}: %{endif}%{message}" telegram 2>&1 |tee telegram-v3.2.5.debug$(date +%Y%m%d).output2

How can I check quickly it does contains?

**Comment by ilya-fedin** [comment]
i just look for `scroll wheel`

**Comment by AndreyKopeyko** [comment]
Now it have "scroll wheel" events but not after 03:02 (notebook was suspended from 03:02 till 03:13)

https://dev.kopeyko.ru/tmp/telegram-v3.2.5.debug20211119.output3

**Comment by ilya-fedin** [comment]
yeah

**Comment by AMV007** [comment]
Steps to reproduce
check scroll working - ok
go to suspend
resume from suspend
mouse scroll not working
Expected behaviour
scroll working

Actual behaviour
scroll not working

Operating system
Ubutnu 20

Version of Telegram Desktop
3.2.4

Installation source
Static binary from official website

**Comment by AndreyKopeyko** [comment]
@ilya-fedin FYI - scrolling via 2-finger gesture on touchpad continues to work fine after wheel scroll fails.
Do you need a Qt debug log with those events?

**Comment by ilya-fedin** [comment]
I don't need any logs, I can't fix this issue since it's not at app level

**Comment by AndreyKopeyko** [comment]
No, you can fix it - just downgrade Qt version used to the working one. 
The problem will be solved.

**Comment by ilya-fedin** [comment]
Well, the problem doesn't seem to be so urgent for such actions

**Comment by ilya-fedin** [comment]
There was a similar issue even before Qt upgrade, #3859. It was existing a big amount of tdesktop lifetime and was closed just because people ignored requests for more information, so maybe they never stopped experiencing the bug, just were so lazy to provide information. So it's something we can live with for years, as the previous experience shows.

**Comment by AndreyKopeyko** [comment]
> Well, the problem doesn't seem to be so urgent for such actions

This is extremely wrong decision.
Maybe it is not urgent for _you_ - but it is important for me and a lot of other people who had faced that wheel scroll issue.

**Comment by AndreyKopeyko** [comment]
> There was a similar issue even before Qt upgrade, #3859. 

I didn't faced that bug, and you definitely can't blame me for not providing debug information.

So, this note must be considered irrelevant to the issue we are discussing now.

**Comment by ilya-fedin** [comment]
>  but it is important for me and a lot of other people

That 'a lot of people' can be a small fraction of userbase, still. There's a lot of issues where people write 'how can't you fix so important bug for years???', the answers is it's not really a big bug. Yeah, for someone it's a big bug in their personal view, but looking more broadly it becomes clear that this is not so important bug for most people, so it has a little priority.

**Comment by ilya-fedin** [comment]
> and you definitely can't blame me for not providing debug information.

No one is blaming you, it's a historical background

> So, this note must be considered irrelevant to the issue we are discussing now.

It's relevant as it shows the bug existed eralier, but maybe it wasn't so reproducible or maybe it was existing in older Qt versions and was fixed at some point, but regressed in Qt 6.x

**Comment by AndreyKopeyko** [comment]
> It's relevant as it shows the bug existed eralier, but maybe it wasn't so reproducible

This time you do have the rock-solid way to reproduce the problem, don't you?

**Comment by ilya-fedin** [comment]
Well, I'm not. I can't reproduce that, so I can't write a minimal reproducible example and can't report this to Qt, so the issue is in dead end (until someone else consider to report that to Qt, moving the issue forward).

**Comment by danger89** [comment]
This escalated quickly. But I fully agree with Andrey here, just down grade qt for now. And we also need to fill a bug report at qt.

**Comment by ilya-fedin** [comment]
Downgrading Qt is not a good idea. Qt 6 has a lot of fixes that never will be backported to 5.x, e.g. fixes for clipboard on X11 & Wayland and Qt 6.3 will have a lot of more important bugfixes, like the fix for #10348. Self-dragging bug is a more important bug than scrolling problems after suspend in my opinion.

**Comment by Aokromes** [comment]
> No, you can fix it - just downgrade Qt version used to the working one. The problem will be solved.

> This escalated quickly. But I fully agree with Andrey here, just down grade qt for now. And we also need to fill a bug report at qt.

upgrading was done to fix bugs, if the amount of bugs and importance it fixes is more than the amount of things it breaks be sure the downgrade will be rejected.

**Comment by danger89** [comment]
Ps. Just for info, in my case the scrolling issue happened without suspend.

Clear. We do not downgrade. 

@AndreyKopeyko could you fill a qt bug report, please? And cross reference to this ticket?

**Comment by 31337-4554551n** [comment]
I can confirm this issue on ubuntu 20.04 and that is without suspend

**Comment by AndreyKopeyko** [comment]
> @AndreyKopeyko could you fill a qt bug report, please? And cross reference to this ticket?

I never filed any bugs to QT so I need some assistance from you. It would be nice if you email me this week later.

**Comment by danger89** [comment]
https://doc.qt.io/qt-6/bughowto.html

**Comment by AndreyKopeyko** [comment]
@Aokromes @ilya-fedin please provide the following information needed to fill a Qt bug

    The name and version number of your compiler
    The version of Qt you are using, and what configure options it was compiled with

**Comment by ilya-fedin** [comment]
> The name and version number of your compiler

GCC 9

> The version of Qt you are using, and what configure options it was compiled with

Qt 6.2.0, the latter depends on where you got tdesktop from (static binary, flatpak and snap - all have different configure flags passed to Qt)

**Comment by AndreyKopeyko** [comment]
I installed deb package provided here.

**Comment by ilya-fedin** [comment]
tdesktop doesn't have deb packages

**Comment by AndreyKopeyko** [comment]
Yes, you are right - it was a binary from tsetup.tar.xz

**Comment by ilya-fedin** [comment]
Here's static binary's Qt configure flags
https://github.com/telegramdesktop/tdesktop/blob/dev/Telegram/build/docker/centos_env/Dockerfile#L744-L761

**Comment by AndreyKopeyko** [comment]
The same problem occurs on Ubuntu-18 

```
kaa@Polina-X270:~$ lsb_release -a 
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.6 LTS
Release:	18.04
Codename:	bionic
kaa@Polina-X270:~$ 
```

**Comment by john-preston** [comment]
Ok, I've made a bug report:

https://bugreports.qt.io/browse/QTBUG-98720

It's a pity not me, not @ilya-fedin can reproduce that, but maybe something could be done from the Qt side.

**Comment by john-preston** [comment]
@AndreyKopeyko

Also I'll try to throw in some more logs in the next beta version so I'll ask for one more clear reproduction-with-logs, like https://dev.kopeyko.ru/tmp/telegram-v3.2.5.debug20211119.output3 was, but for 3.2.6 beta version (that'll be out later today or in a couple of days).

**Comment by belousoves** [comment]
Have same problem. 
version 3.2.4 on linux Mint 20.2.
Scrolling using mouse wheel never works, restarting telegram does not help.
Scrolling by PgUp/PgDn works.

**Comment by john-preston** [comment]
@belousoves BTW Linux build 3.2.6 beta should have the additional logs. I doubt they'll help fixing the issue, but who knows.. at least I'll add them to the Qt bug report.

**Comment by rromrrom** [comment]
Fedora Linux 35. Gnome. Telegram 3.2.5. I have the same issue: Mouse wheel scrolling doesn't work while both touchpad two-fingers and pgup/pgdown works fine. Annoying bug.

**Comment by john-preston** [comment]
@AndreyKopeyko @belousoves @rromrrom @danger89 @31337-4554551n @AMV007 @thenetengineer @jprando 

1. I want to get new set of logs from 3.2.6 beta version (or later) obtained with such command:
```
QT_LOGGING_RULES='*.debug=true' QT_MESSAGE_PATTERN="[%{time}] %{if-category}%{category}: %{endif}%{message}" ./Telegram 2>&1 |tee telegram-v3.2.6.debug$(date +%Y%m%d).txt
```

I want such log (`telegram-v3.2.6.debug[smth].txt`) from launch, through the suspend, till the bug is reproduced and you try to scroll several times without it working.

2. The question from the Qt bug tracker:

![image](https://user-images.githubusercontent.com/17900494/144209543-ce31fbb8-52a2-497f-8d87-fcbcccf12dce.png)

**Comment by belousoves** [comment]
@john-preston 
On beta 3.2.6 bug is not reproducible. Scroll works fine.

**Comment by john-preston** [comment]
It's (nearly) impossible. Maybe just didn't try long enough.. Anyway, if it reproduces again, please send the log.

**Comment by john-preston** [comment]
@AndreyKopeyko @belousoves @rromrrom @danger89 @31337-4554551n @AMV007 @thenetengineer @jprando

Do you have this issue with a USB mouse or a Bluetooth mouse?

**Comment by rromrrom** [comment]
> @AndreyKopeyko @belousoves @rromrrom @danger89 @31337-4554551n @AMV007 @thenetengineer @jprando
> 
> Do you have this issue with a USB mouse or a Bluetooth mouse?

On my side it is Bluetooth mouse.

**Comment by vjrj** [comment]
On my side USB mouse.

**Comment by belousoves** [comment]
On my side it is USB wireless(but not Bluetooth) mouse with keyboard.

**Comment by 0xffh** [comment]
Same on ubuntu 20.04.3 and wireless mouse. The touchpad is fine. Telegram version 3.2.2

**Comment by danger89** [comment]
In my case it's all wired. Wired keyboard, wired mouse.

**Comment by belousoves** [comment]
Telegram 3.2.8 beta, bug reproduces again. Scroll never works, no need to suspend or doing something else. Scrolling does not works just after telegram start.
Attaching lo
[telegram-v3.2.8.debug20211204.txt](https://github.com/telegramdesktop/tdesktop/files/7654456/telegram-v3.2.8.debug20211204.txt)
g.

**Comment by john-preston** [comment]
@belousoves But this should be something else then, because in the initial report scroll works and then stops. This might be completely different (or might not) from the case when it simply doesn't work at all.

**Comment by john-preston** [comment]
@belousoves And the log is very small and doesn't contain any of those scroll input event log lines at all :(

**Comment by mwoodson** [comment]
Adding some more info on what I've seen and reproducer.  I have Fedora 35 + KDE.  I also have VFIO VM with a USB switch. 

VFIO is a VM that runs passthrough to certain devices (like usb card, hard drive, gpu) to a VM for direct access.  

My config:

mouse/keyboard -> usb switch -> 2 different USB ports on the computer.  
USB1 = VM USB 
USB2 = PC USB (never gets passed through)

When I switch to my VFIO VM, I switch the USB switch to use USB1.  Windows is happy.  When I turn off VM and come back to the Linux PC, the mouse works, but the scrolling in telegram is broken.  All other apps seem to work fine (chrome, for example).


I also noticed that If I plug a mouse into USB2 (which is always connected to the linux PC, never passed through to the VM), scrolling still works with that mouse, but will be broken on the mouse that is passed through.

Let me be clear, I have a complex setup.  I don't expect this to be the reproducer.  But having the mouse get disconnected and reconnected seems to be what is causing the loss of scrolling for me.  This seems similar to the supsend/hibernation and is not getting picked back up by QT.


I figure the info doesn't hurt.

**Comment by joeytwiddle** [comment]
How are you running older versions? I would like to run an older version until this bug is fixed.

`snap` and `flatpak` only seem to offer me the latest version.

I found old v1.2.6 but when I give it my phone number, it responds "Internal Server Error".

I tried installing the [3.1.13 release](https://github.com/telegramdesktop/tdesktop/releases/tag/v3.1.13) from this repo, but when I run it:

- "Integrate and Run" fails with `ERROR: appimage_register_in_system : Missing Desktop Entry`
- and "Run Once" did nothing (process running but no window, no taskbar icon and no console output)
- (I haven't even deleted the `Updater` executable yet, but that was going to be the plan...)

Is there another way I could try? Thanks in advance.

**Comment by ilya-fedin** [comment]
There's a coincidence that versions before update to Qt 6 no longer working due to server API update

**Comment by AndreyKopeyko** [comment]
> @AndreyKopeyko @belousoves @rromrrom @danger89 @31337-4554551n @AMV007 @thenetengineer @jprando
> 
> Do you have this issue with a USB mouse or a Bluetooth mouse?

I worked for about 2 weeks with 3.2.5 without restarting it, and found some new stranges in wheel scroll behaviour - from time to time it cures and fails back later.

It happens so that I use 5 different devices to perform wheel scrolling
- Logitech M235 wireless mouse with nanoreceiver plugged into usb-2 port on Lenovo docking station
- "trackpoint up&down +3rd button pressed" on Lenovo compact usb keyboard 
- Microsoft Sculp bluetooth mouse
- 2-finger touchpad gesture on my Thinkpad T460 
- "trackpoint up&down +3rd button pressed" on Thinkpad body

First & second devices are used in office; other are used at my home (touchpad and trackpoint-on-the-body aren't available at the office because laptop sits in the dock with lid closed). So, in any condition I can use 2 different scroll devices minimum.

They all fail randomly and independently
- everything works fine right after telegram is started
- if Logitech mouse wheel scroll fails, trackpoint+3rdButthon mainly continues to work. But sometimes it fail too.
- if bluetooth mouse wheel scroll fails, touchpad continues to work. But it can fail too.
- if touchpad gesture scroll fails, trackpoint+3rdButthon usually can do the scrolling work. But may fail too

After 2-3 days I suddenly find out that wheel scrolling can "repair itself" on some devices. But next time it can fail back... Or can continue to work.

I was unable to figure any pattern it their behaviour. Just try and see if this kind of wheel scroll works now or not.

The only rule discovered is that status change happens when note is resumed from sleep, or when screen is unlocked. 

I was unable to grep smth interesting from OS logs.

Hope this details may help you to understand the root cause.

Sorry for my long silent period - I was busy with work problems. Now I am heading to install & test 3.3.0 binary you've released few days ago.

**Comment by pfortin** [comment]
WEIRD: mouse quit scrolling a few days ago on v3.3. Because this wireless mouse shares the USB dongle with a keyboard, and wanting to use that pair for a Pi4B I just ordered, I bought a new wireless mouse. Plugged it in and telegram scrolling works again. Previous mouse now works too -- without restarting Telegram... (Twilight Zone)

**Comment by DC7NTP** [comment]
i have literally had this problem for ages now. even in version 3.3 it happens quite often. the application appears as if it is being dragged around using left mouse button and Alt-key, even though no keys or buttons were pressed. I have the four-way arrow everywhere on the application window and no amount of clicking and pressing keys works to get rid of it. Scrolling using a mouse wheel of two-finger gesture of course does not work. and moving the mouse or pointer around moves the application window around without any keys or buttons being pressed. Only force closing and restarting telegram does work for a little while. I am running Arch Linux with XFCE desktop

**Comment by ilya-fedin** [comment]
> i have literally had this problem for ages now. even in version 3.3 it happens quite often. the application appears as if it is being dragged around using left mouse button and Alt-key, even though no keys or buttons were pressed.

This sounds like #10348 rather than this issue

**Comment by DC7NTP** [comment]
seems more like i am actually having both problems. Will now try to see if the suggested fix Enable "System Windows Frame" will do anything. Does not seem like it, sorry.

**Comment by pfortin** [comment]
Another weird problem (blurred images; videos OK) cropped up with Telegram over the last 12-18 hours. I managed to resolve thes issue by:
1. Closing Telegram
2. Finding a still running Telegram process
3. Terminating that background process.
Like this:
$ ps aux | grep -i telegram
pfortin  2573169  1.5  1.2 2783736 402960 pts/7  Sl   Dec07 197:34 /home/pfortin/Downloads/Telegram/Telegram/2.5.7/Telegram/Telegram -noupdate
$ kill 2573169
$ Telegram &
... it's running fine again, for now...  Maybe this will help the mouse problem.

**Comment by ilya-fedin** [comment]
Why do you kill it instead of quitting via tray icon?

**Comment by AndreyKopeyko** [comment]
Finally I tried 3.3.0 and it demonstrates the same behaviour 
- mouse scroll fails after laptop resumes from sleep
- it fails independently to all scrolling devices I use
- it can repair back for some device - and I still can't determine any regularity...

@ilya-fedin @john-preston shall I create another debug log from 3.3.0 and send it to you?

**Comment by pfortin** [comment]
> Why do you kill it instead of quitting via tray icon?
Valid question...  I was curious if there was something hanging around; found it and killed it from where I found it.  Brain fart: Forgot that some apps leave a process running when you "close" them.

**Comment by snorlax212** [comment]
xubuntu 20.04, telegram version 3.3. also encounter this bug, it appears randomly, mostly on startup. relaunch helps.

**Comment by vazir** [comment]
2 mouses connected at the same time, both Razer via BT, 1-atheris, 2-basilisk X. Basilisk - lost scrolling, Atheris still scrolls. Restarting Telegram restore scrolling in the chat... For a while

**Comment by prannonpendragas** [comment]
Running Manjaro Linux with Telegram desktop 3.3. Having the same intermittent scroll wheel issue.

- Using a Razer OrochiV2 wireless mouse, with bluetooth.
- Scrolling works for a while after starting the app.
- If the laptop is locked for a while (or if I leave the mouse unused for a while - maybe the bluetooth sleeps temporarily) then scrolling will not work.
- If I ctrl-q the telegram app and then reload the window, scrolling works again.

**Comment by bast1aan** [comment]
Telegram Desktop installed with Snap on Ubuntu 18.04 and 20.04, same issue. It only appears after a while, not immediately.

**Comment by Aokromes** [comment]
maybe
https://newbedev.com/how-to-disable-auto-power-off-of-usb-devices-like-usb-mouse
can work as workaround

**Comment by Virusmater** [comment]
Currently I have a situation when wireless mouse scroll stopped working, but laptop touchpad and trackpad are working

**Comment by belousoves** [comment]
I found out scroll stops working after telegam update sometimes. 
If I reboot system, and start telegram arain, scroll works fine on same tg version.

**Comment by belousoves** [comment]
PS restarting telegram witout rebooting does not fix problem. Reboot only.

**Comment by ezze** [comment]
Not sure my comment will be usable but I want to mention that in my case (despite others) restarting Telegram fixes the issue without need to reboot in Ubuntu 20.04 (KDE).

**Comment by Aokromes** [comment]
try https://github.com/telegramdesktop/tdesktop/issues/17254#issuecomment-1008328640

**Comment by ernado** [comment]
Ubuntu 21.10

Telegram is the only software with broken scroll on my system.

Restarting Telegram helps, but this is very temporary workaround.

It is very sad :( Scroll is broken for me for several month already.


@ilya-fedin @john-preston gentle ping

Can I manually downgrade Qt or do something different to fix this issue?

**Comment by ilya-fedin** [comment]
No, Qt is a part of Telegram binary.

**Comment by ernado** [comment]
I mean recompile it manually or something like that. Is it trivial or new functionality from Qt is used?

**Comment by ilya-fedin** [comment]
I think build with Qt 5 is broken with official instructions currently, but theoretically it's possible if someone fixes cmake scripts (personally I'm not interested in that)

**Comment by ilya-fedin** [comment]
Guys, if you want to speed up this, it's better if you would comment the Qt bug rather than this issue

**Comment by ilya-fedin** [comment]
Right now Qt bug report looks like it's affecting only one person

**Comment by ernado** [comment]
Ok, I've reproduced this bug with turning off and on my usb hub multiple times.
```
qt.qpa.input.events: (tdesktop) scroll wheel compress XCB_INPUT_MOTION NO
qt.qpa.events: Event | XInput Event(XCB_INPUT_MOTION) | sequence: 4822
qt.qpa.input.events: (tdesktop) scroll wheel got in XCB_GE_GENERIC
qt.qpa.input.events: (tdesktop) scroll wheel got event 6 in xi2HandleEvent
qt.qpa.input.events: (tdesktop) scroll wheel got input_motion in xi2HandleEvent
qt.qpa.input.events: XI2 mouse motion 420,143, time 318007192, source MouseEventNotSynthesized
qt.qpa.events: Event | XInput Event(XCB_INPUT_BUTTON_PRESS) | sequence: 4822
qt.qpa.input.events: (tdesktop) scroll wheel got event 4 in xi2HandleEvent
qt.qpa.events: Event | XCB_XKB_* event(85) | sequence: 4822
qt.qpa.events: Event | XInput Event(XCB_INPUT_BUTTON_RELEASE) | sequence: 4822
qt.qpa.input.events: (tdesktop) scroll wheel got event 5 in xi2HandleEvent
qt.qpa.events: Event | XCB_XKB_* event(85) | sequence: 4822
qt.qpa.input.events: (tdesktop) scroll wheel check XCB_INPUT_MOTION compress
qt.qpa.input.events: (tdesktop) scroll wheel compress XCB_INPUT_MOTION NO
qt.qpa.events: Event | XInput Event(XCB_INPUT_MOTION) | sequence: 4822
qt.qpa.input.events: (tdesktop) scroll wheel got in XCB_GE_GENERIC
qt.qpa.input.events: (tdesktop) scroll wheel got event 6 in xi2HandleEvent
qt.qpa.input.events: (tdesktop) scroll wheel got input_motion in xi2HandleEvent
qt.qpa.input.events: XI2 mouse motion 420,143, time 318007270, source MouseEventNotSynthesized
qt.qpa.events: Event | XInput Event(XCB_INPUT_BUTTON_PRESS) | sequence: 4822
qt.qpa.input.events: (tdesktop) scroll wheel got event 4 in xi2HandleEvent
qt.qpa.events: Event | XCB_XKB_* event(85) | sequence: 4822
qt.qpa.events: Event | XInput Event(XCB_INPUT_BUTTON_RELEASE) | sequence: 4822
qt.qpa.input.events: (tdesktop) scroll wheel got event 5 in xi2HandleEvent
qt.qpa.events: Event | XCB_XKB_* event(85) | sequence: 4822
qt.qpa.input.events: (tdesktop) scroll wheel check XCB_INPUT_MOTION compress
qt.qpa.input.events: (tdesktop) scroll wheel compress XCB_INPUT_MOTION NO
qt.qpa.events: Event | XInput Event(XCB_INPUT_MOTION) | sequence: 4822
qt.qpa.input.events: (tdesktop) scroll wheel got in XCB_GE_GENERIC
qt.qpa.input.events: (tdesktop) scroll wheel got event 6 in xi2HandleEvent
qt.qpa.input.events: (tdesktop) scroll wheel got input_motion in xi2HandleEvent
```

I see in logs scroll events, but nothing happens in telegram.

Are you sure that it is 3rd party issue?

**Comment by john-preston** [comment]
@ernado Can you please reproduce it while collecting logs like stated above:

https://github.com/telegramdesktop/tdesktop/issues/17254#issuecomment-983459527

Can you please provide a full log after that along with a couple of timestamps - when the scroll definitely was working (early timestamp) and when it definitely didn't, although the events should've been sent (later timestamp).

**Comment by ernado** [comment]
Sorry for flickering in video:

https://user-images.githubusercontent.com/866677/150084316-851bc340-fa05-4734-ad53-e4dfb4c97a5b.mp4

I'll post logs for repro soon

**Comment by ernado** [comment]
[telegram-v3.2.6.debug20220119.txt](https://github.com/telegramdesktop/tdesktop/files/7894980/telegram-v3.2.6.debug20220119.txt)

So issue is happening after `register QInputDevice`, when USB hub is getting online and connection mouse again.

So it looks like device registration is somehow related.

Also it not 100% reproducible, sometimes it just works after hub on-off-on cycle, probably some race (in QT?).

**Comment by john-preston** [comment]
@ernado So, in this log, when did it work and when it stopped?

**Comment by ernado** [comment]
@john-preston
[telegram-scrolling-log.txt](https://github.com/telegramdesktop/tdesktop/files/7900953/telegram-scrolling-log.txt)

```
[2022-01-20T01:29:43] <- OK
[2022-01-20T01:29:49] qt.qpa.screen: output "DP-2" is connected and enabled; was fake: ":1.0"
My USB hub is in my monitor, so screen is re-connected as well as other devices like mouse.
Scroll is not working from that moment.
[2022-01-20T01:29:56] <- Scroll attempt, not working
```

**Comment by john-preston** [comment]
@ernado Ok, I've linked this information in the Qt bug report. I'm not sure I'm qualified (have enough time?) to debug it myself, I still hope Qt will fix this.

**Comment by satsumah** [comment]
This bug occurs while switching devices on a hardware KVM as well. Ubuntu 20.04, Telegram version 3.4.3.

**Comment by billouetaudrey** [comment]
Hello, I have same issue, nothing help. Happens only with Telegram too :'( 

Linux Mint xfce

**Comment by kbokh** [comment]
The same - Pop!_OS 21.10

**Comment by sufehmi** [comment]
Hi, I got same issue here, Telegram Desktop 3.4.3 on Snap, Ubuntu Mate 20.04, Logitech MX Anywhere 3.

Restarting Telegram does not work, have to restart laptop.

Scrolling via Touchpad still works.

**Comment by sprinter11** [comment]
Have the same issue. Before Telegram Desktop v 3.4.8 the mouse reconnection trick worked. Updating to v 3.4.8 makes it worse, now only restarting the app helps.
Ubuntu 20.04.3 LTS

**Comment by max246** [comment]
Same issue With Linux mint, I am using a kvm to swtich now and then , might that the trigger. Any solution? It is annoying to close and open the app each time

**Comment by ilya-fedin** [comment]
I believe comments like 'same issue' are enough, they're not really useful.

**Comment by feeela** [comment]
This bug was introduced for me only after the update to v3.4.8 @ Ubuntu 20.04.

**Comment by max246** [comment]
> I believe comments like 'same issue' are enough, they're not really useful.

I am happy to provide more logs or run a debug version, just need to be told what I can provide for extra information!

**Comment by Aokromes** [comment]
go report https://bugreports.qt.io/browse/QTBUG-98720

**Comment by sufehmi** [comment]
Okay guys, let's do this: https://bugreports.qt.io/browse/QTBUG-98720?focusedCommentId=642944&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-642944[https://bugreports.qt.io/browse/QTBUG-98720?focusedCommentId=642944&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-642944](url)

**Comment by john-preston** [comment]
It is a third party issue, but I think it is a good idea to track it here as well, sounds serious enough. Please no flood here.

**Comment by feeela** [comment]
@ilya-fedin

From my end-user point-of-view this is a Telegram issue and I am not interested in some flimsy excuses.

Telegram right now is the only software installed on my system, that shows that kind of bug. If **you chose third-party libs with bugs** in your application, then **of course you will have to handle them**.

Right now, you are just ignoring everyone, which is why more and more people commented with “same issue” – because you were just not willing to react in any sensible way.

It is surely not the users responsibiliy to fix your broken dependencies or even figure out that this was caused by a broken dependency. And you even bundled the QT-deps into Telegram as stated above (“tdesktop doesn't use your system Qt, it has Qt 6.2 embedded in the binary”) – as an end-user we don't even have the chance of using another version of that dependency.

And then closing an issue with the obvious attitude of “wont fix”, is – from my point of view – snobbish and ignores your users who might have used your app a bit longer. Sir, you are effectivly losing users here and you just don't care at all.

**Comment by ilya-fedin** [comment]
It's just about of the ways of handling issues. If a project has a policy of closing third-party issues and tracking only in the dependency's bugtracker, it has full right to do so. If you don't like that, well, it's your problems.

**Comment by john-preston** [comment]
@feeela Sorry, but we'll still have to wait for Qt to fix that 😔

**Comment by ilya-fedin** [comment]
> And you even bundled the QT-deps into Telegram as stated above (“tdesktop doesn't use your system Qt, it has Qt 6.2 embedded in the binary”) – as an end-user we don't even have the chance of using another version of that dependency.

It's your choice how to install tdesktop. You can install tdesktop from system repositories and it will use your system Qt. If you choose official universal packages, of course they would have bundled Qt, you can't have a relocatable binary that would work on all the Linux distributions, just like all other software having universal packages. This is no more different than all other software on your system, but you've chosen to install tdesktop from universal package unlike all other software. That's only your choice.

**Comment by ernado** [comment]
Sorry for bumping this issue for everyone, but I will cross-post:

On Linux host:

1. Scrolling works
2. Unplug mouse
3. Plug mouse
3. Scrolling is not working

Looks like there is much simpler reproducer (at least it works for me with high success rate), so no system sleep needed.

**Comment by ilya-fedin** [comment]
> If **you chose third-party libs with bugs** in your application, then **of course you will have to handle them**.

I've never seen a bugless library... Every software, every library has bugs.

**Comment by john-preston** [comment]
@ernado You better write this (and add something like "I'm ready to help if needed, to collect some logs or anything.") in QTBUG linked above.

**Comment by ernado** [comment]
@john-preston yep, it was the first thing I did

Thank you for filing the initial but, hope it will be fixed soon

**Comment by Aokromes** [comment]
> @ilya-fedin
> 
> From my end-user point-of-view this is a Telegram issue and I am not interested in some flimsy excuses.
> 
> Telegram right now is the only software installed on my system, that shows that kind of bug. If **you chose third-party libs with bugs** in your application, then **of course you will have to handle them**.
> 
> Right now, you are just ignoring everyone, which is why more and more people commented with “same issue” – because you were just not willing to react in any sensible way.
> 
> It is surely not the users responsibiliy to fix your broken dependencies or even figure out that this was caused by a broken dependency. And you even bundled the QT-deps into Telegram as stated above (“tdesktop doesn't use your system Qt, it has Qt 6.2 embedded in the binary”) – as an end-user we don't even have the chance of using another version of that dependency.
> 
> And then closing an issue with the obvious attitude of “wont fix”, is – from my point of view – snobbish and ignores your users who might have used your app a bit longer. Sir, you are effectivly losing users here and you just don't care at all.

no, it was not attitude of “wont fix”.

**Comment by satorym** [comment]
Same issue. First time was approximately 4 months ago, ignored it bacause restart is fixing problem. Doesn't know how to reproduce, it happens when telegram is minimized for long time but not always.

Kubunu 20.04.2
Qt: 5.12.8
Kernel: 5.13.0-27

**Comment by ztane** [comment]
Yes, me too can confirm, Ubuntu 20.04. I've never experienced it before when using the .deb. I only started experiencing it now that I switched from the outdated .deb package to the snap one. Wheel always works initially but it stops at some point. It is really annoying, because it worked always for 5 years.

**Comment by ztane** [comment]
Since it is now a snap then whoever builds the package is the one who bundles in the relevant Qt libs. Not pointing fingers anywhere since I do not know who builds the snaps, but Ubuntu Qt versions should not matter directly?

**Comment by john-preston** [comment]
@ztane Yes, I build the snaps on launchpad.

**Comment by Popolon** [comment]
If it can help, I first noticed this bug when I unplugged/replugged mouse+keyboard from a little USB hub, because I use it on several computers. The bug occurred then, when this (unplug/plug) was done during a Telegram-Desktop session. I just had to stop Telegram and restart it again, to solve the issue. Today, with version 3.4.8 (I use ArchLinux, with frequent update) mouse wheel doesn't work at all, even if I stop and restart Telegram.

**Comment by Popolon** [comment]
oops this seem to be solved, that strange, it worked in some other applications including firefox and didn't work on Telegram Desktop, even after restarting. I plugged it in a RISC-V based Deiban bookworm/sid (same 3.4.8 version), and it worked fine, I replugged it then in the x86 Archlinux and work too, so at least the bug of hotplug is resolved.

**Comment by RamilMus** [comment]
Stop telegram, unplug mouse. Plug back, start telegram, worked on Ubuntu 20.

**Comment by tanyeun** [comment]
> Stop telegram, unplug mouse. Plug back, start telegram, worked on Ubuntu 20.

this is a quick fix, haha, thanks!

**Comment by bdcuck** [comment]
I have the same issue on version 3.5 on SalientOS. Restarting Telegram fixes it without need to unplug my mouse. This seems to happen randomly after a while. I do swap my mouse to another PC pretty often, yet I can't reproduce this by simply unplugging and plugging the mouse back in while Telegram is running.

**Comment by gergely-42droids** [comment]
Same issue on Telegram v 3.5.2 (08.02.22), Ubuntu Mate 21.04 (Kernel Linux 5.11.0-49-generic x86_64)
Telegram scroll breaks after about 2 days. Restarting Telegram solves the problem.
Hibernation/suspension might be the cause, but not 100%.

**Comment by tkachen** [comment]
> I have the same issue on Telegram version 3.5, Qt 5.15.2 on SalientOS. Restarting Telegram fixes it without need to unplug my mouse. This seems to happen randomly after a while. I do swap my mouse to another PC pretty often, yet I can't reproduce this by simply unplugging and plugging the mouse back in while Telegram is running.
> 
> Desktop PC and I never suspend/hibernate it btw.

Exactly same behavior. 

Telegram 3.5 on Manjaro. Can't reproduce by anything - problem just appears at some moment. Restart of Telegram without unplugging mouse fixes the problem in 100% of cases. Never used suspension or hibernation.

**Comment by tkachen** [comment]
> adding "i have the same bug" here won't help to be fixed, upvote at https://bugreports.qt.io/browse/QTBUG-98720

Why though? That issue has ref to this page. Besides people with this problem more likely will find this page. By describing different conditions and temporary fixes we can provide more info for people with problem. QT team already marked this as critical bug and can open this thread for info when they decide to solve it.

**Comment by Aokromes** [comment]
> > adding "i have the same bug" here won't help to be fixed, upvote at [bugreports.qt.io/browse/QTBUG-98720](https://bugreports.qt.io/browse/QTBUG-98720)
> 
> Why though? That issue has ref to this page. Besides people with this problem more likely will find this page. By describing different conditions and temporary fixes we can provide more info for people with problem. QT team already marked this as critical bug and can open this thread for info when they decide to solve it.

because they are who must fix it, posting here won't adding preasure qt devs to fix it because they very likely don't follow this issue.

**Comment by bast1aan** [comment]
Might worth noting the bug appears to me on Kubuntu 18.04 with telegram-desktop 3.4.3 <b>as a Snap app</b>, yet the bug does not appear on my other laptop running Kubuntu 20.04 with telegram-desktop 3.4.2 installed <b>as a Flatpak app</b>

**Comment by ilya-fedin** [comment]
I guess what makes the difference here is different hardware

**Comment by bast1aan** [comment]
My guts says it has more to do with how snap/flatpak handles it differently like one triggers the bug, the other does not, than it has to do with hardware. The same kind of mouses are attached on both, anyway, and both are running intel USB chipsets.

**Comment by ilya-fedin** [comment]
> My guts says it has more to do with how snap/flatpak handles it differently like one triggers the bug, the other does not

That's unlikely since snap/flatpak has nothing to do with mouse

**Comment by bast1aan** [comment]
I've seen more of these kind of bugs in Snap. Like keyboard layout settings are incorrect, and GUI styles are missing.

**Comment by ilya-fedin** [comment]
Well, keyboard layouts are handled client-side with libraries like xkbcommon or ibus. GUI styles, is something impossible to achieve in confined environments like flatpak or snap since styles depend on libraries version (e.g. in Qt styles are C++ libraries), so using styles from the host is impossible. But mouse scrolling is just an event sent through X socket, so flatpak/snap has nothing to do with that.

> Also, I don't have this scrolling bug in any other application (snap, flatpak, deb, gtk, qt, whatever) on neither device.

Telegram uses Qt 6, are you sure 'any other application' uses Qt 6, too?

**Comment by vsinitsyn** [comment]
> But mouse scrolling is just an event sent through X socket

Correct. And the most weird part here is that when the bug triggers, it affects mouse only. I can still scroll within Telegram using touchpad on my Snap/Kubuntu 20.04 system, even if touchpad and mouse wheel should generate the same event.

**Comment by ilya-fedin** [comment]
> even if touchpad and mouse wheel should generate the same event under X.

Qt sees the difference between the wheel and touchpad, so apparently they don't

**Comment by bast1aan** [comment]
Fair enough. I will see coming days how the flatpak app will do. Feel free to clean up the discussion if you don't like the clutter :)

**Comment by Aokromes** [comment]
> Fair enough. I will see coming days how the flatpak app will do. Feel free to clean up the discussion if you don't like the clutter :)

you may want to give a try to latest qbittorrent, they updated to QT 6.

**Comment by danger89** [comment]
Today I have the same mouse scrolling again, using [Telegram static build binary](https://telegram.org/dl/desktop/linux) for Linux. Page-up and page-down still works. But scrolling is not working in the chat windows, but the scroll-wheel also stop responding in the people/group list on the left-side.

I'm **_NOT_** using the snap package.

**Comment by bast1aan** [comment]
> flatpak app (3.5.2) and will run it on the Kubuntu 18.04 laptop where the bug is present, and will report back, so maybe it can indicate if it has something to do with Snap or is a hardware/kernel/xorg version related problem.
Can confirm this bug appears on 3.5.2 / flatpak as well. The mouse stops working, yet the touchpad two-finger scrolling still works.
> you may want to give a try to latest qbittorrent, they updated to QT 6.
Unfortunately I can't find a qbittorrent version using Qt6 that works on my system. 4.4.1 from Flatpak still uses Qt5, the appimage doesn't work due to glibc problems.

**Comment by ilya-fedin** [comment]
It's time to switch to Arch

**Comment by marcumen** [comment]
Same thing here on Ubuntu, happens from time to time. I can't find a way to reproduce the issue.

**Comment by Mayurifag** [comment]
Same issue on community/telegram-desktop 3.5.2-1 in archlinux. If you unlock pc and want to use telegram, scroll is broken on linux.

**Comment by Aokromes** [comment]
https://github.com/telegramdesktop/tdesktop/issues/17254#issuecomment-1038076112

**Comment by anteklincic** [comment]
> It's time to switch to Arch

Why? I have the same bug on vanilla Arch, using telegram-desktop from official Arch repo. I don't think this has to do with putting computer to sleep, this issue happens all the time to me. I will try to record my mouse movements with a camera, this bug is easy to reproduce on my side.

**Comment by ilya-fedin** [comment]
> Why?



> Unfortunately I can't find a qbittorrent version using Qt6 that works on my system. 4.4.1 from Flatpak still uses Qt5, the appimage doesn't work due to glibc problems.

**Comment by anteklincic** [comment]
> > Why?
> 
> > Unfortunately I can't find a qbittorrent version using Qt6 that works on my system. 4.4.1 from Flatpak still uses Qt5, the appimage doesn't work due to glibc problems.

You can remove your comment, I will do the same, sorry, I am new here.

**Comment by DavidBoltuev** [comment]
I have reproduced this bug. I wanted to try out the new Logitech MX Master 2S mouse and got the same bug, but all the other mice worked well. Restarting the computer helped fix

**Comment by RC4347** [comment]
I switched to beta version then update fixed!

**Comment by AbhiTheModder** [comment]
### One better fix I tried and working perfectly even with windows :
Close any application behind the telegram desktop app which responds to scroll behavior like in desktop explorer responds to scrolling so while windows explorer open you can't use scroll in telegram desktop similarly in linux or any system terminals are more respond able to scroll behavior and telegram as a focusable messenger let's background app to use that feature instead of using itself so that user can focus also on their tasks so better close any terminal command line promt etc for tg to be work perfectly

**Comment by Rav3nPL** [comment]
OS: Arch linux
TG installed via AUR (3.6.1)
Scroll via mouse wheel stoping working after hibernation.
Restart of TG helps.

**Comment by Aokromes** [comment]
https://github.com/telegramdesktop/tdesktop/issues/17254#issuecomment-1038076112

**Comment by ocosta** [comment]
Just out of curiosity, I launched Telegram from a terminal on Fedora 35, and these are the last lines that appear before it stops processing mouse wheel scroll:

```
qt.gui.imageio.jpeg: Corrupt JPEG data: premature end of data segment
QTextCursor::setPosition: Position '-1' out of range
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
qt.gui.imageio.jpeg: Corrupt JPEG data: premature end of data segment
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
```
Before these, there are dozens of

`
qt.svg: Error while inflating gzip file: SVG format check failed
`
But they are not the root-cause (they do not trigger the problem).

**Comment by Wiedzmin89** [comment]
I have the same problem. Scrolling doesn't work if mouse was connected after starting telegram.
If you first connect the mouse, and then run the application, everything works

**Comment by Aokromes** [comment]
https://github.com/telegramdesktop/tdesktop/issues/17254#issuecomment-1038076112

**Comment by ArsenyMironov** [comment]
> ### One better fix I tried and working perfectly even with windows :
> Close any application behind the telegram desktop app which responds to scroll behavior like in desktop explorer responds to scrolling so while windows explorer open you can't use scroll in telegram desktop similarly in linux or any system terminals are more respond able to scroll behavior and telegram as a focusable messenger let's background app to use that feature instead of using itself so that user can focus also on their tasks so better close any terminal command line promt etc for tg to be work perfectly

This worked for me in POP_OS. Opened a new workspace, opened Telegram there, scrolling started to work again. The bug happened after suspending the laptop.

Thanks!

**Comment by Wirtos** [comment]
I've found a way to reproduce the issue consistently, checked on KDE and XFCE:
Drag the windows and then immediately try to scroll chats list, then it breaks. Might record a video later.

**Comment by ilya-fedin** [comment]
> Drag the windows and then immediately try to scroll chats list, then it breaks.

It's another issue, #10348

**Comment by Wirtos** [comment]
> > 
> 
> It's another issue, #10348

Are you sure? The bug I encounter has nothing to do with snapping but rather it switches scrolling into window grab mode so scrolling doesn't work properly afterwards

**Comment by ilya-fedin** [comment]
> Are you sure?

Yeah, the person who opened the issue observed it when snapping the window, but the bug happens not only when snapping, but also after moving/resizing the window.

**Comment by pedrokpp** [comment]
I also have the same issue but it only occurs when I keep Telegram opened for a few hours (usually 2 to 3 hours), scrolling just stops working

The only way to fix for me is restarting the app

Installed with `pacman`:

```zsh
$ uname -a            
Linux archlinux 5.17.1-arch1-1 #1 SMP PREEMPT Mon, 28 Mar 2022 20:55:33 +0000 x86_64 GNU/Linux
```

```zsh
$ pacman -Qi telegram-desktop
Name            : telegram-desktop
Version         : 3.6.1-1
Description     : Official Telegram Desktop client
Architecture    : x86_64
URL             : https://desktop.telegram.org/
Licenses        : GPL3
Groups          : None
Provides        : None
Depends On      : hunspell  ffmpeg4.4  hicolor-icon-theme  lz4  minizip  openal  ttf-opensans  qt6-imageformats  qt6-svg  qt6-wayland  qt6-5compat  xxhash  glibmm  rnnoise  pipewire
                  libxtst  libxrandr  jemalloc  abseil-cpp  libdispatch
Optional Deps   : webkit2gtk: embedded browser features [installed]
                  xdg-desktop-portal: desktop integration
Required By     : None
Optional For    : purpose
Conflicts With  : None
Replaces        : None
Installed Size  : 70,81 MiB
Packager        : Sven-Hendrik Haase <svenstaro@gmail.com>
Build Date      : qua 16 mar 2022 16:40:12
Install Date    : sáb 26 mar 2022 09:08:52
Install Reason  : Explicitly installed
Install Script  : No
Validated By    : Signature
```

**Comment by Aokromes** [comment]
https://github.com/telegramdesktop/tdesktop/issues/17254#issuecomment-1038076112

**Comment by john-preston** [comment]
We're locking the issue due to the fact that unfortunately it can't be resolved here.

If you have some valuable details about this issue, please report them in the upstream bug report:

https://bugreports.qt.io/browse/QTBUG-98720

**Comment by ilya-fedin** [comment]
Given that Qt's X11 support degrades and it seems they have no people with enough knowledge to fix bugs like this one or #10348 on X11, we're trying to remove GNOME from blacklist for operating with Wayland backend again (it's the only blacklisted environment). Last time we tried that, there were a lot of reports about regressions as migrating from X11 backend to Wayland backend with Qt is basically exchanging one bugs with the others as both backends have various user-noticable bugs. So, this time we're trying to make Wayland the default only for new installations, so these bugs appear as result of manual intervention (like installing the application on new system installation, switching from X11 session to Wayland or clearing application data) rather than a regression after automatic update.

So, if you're using any Wayland-capable environment (like Ubuntu 22.04 with non-NVIDIA hardware) and you aren't afraid of exchangiтg the current bugs with some other ones, you can update to beta version 4.0.3 and clear application data or remove `nowayland` file from `tdata` directory in application data directory (you can open it with the `viewlogs` [cheat code](https://github.com/telegramdesktop/tdesktop/wiki/Cheat-Codes)).

**Comment by DrjavaB** [comment]
telegram desktop
v 4.0.2
OS: ubuntu 20.04 lts
telegram desktop mouse scroll not working! neither in groups nor in the settings menu

**Comment by ilya-fedin** [comment]
Ok, apparently the thread should be locked again to avoid noise

**Comment by ilya-fedin** [comment]
Everyone who want the fix right now: feel free to switch to Wayland and follow what is described in https://github.com/telegramdesktop/tdesktop/issues/17254#issuecomment-1198891921

**Comment by eyalroz** [comment]
@ilya-fedin : Don't you mean 6.6.0? That's what it says on the [Qt bug page](https://bugreports.qt.io/browse/QTBUG-98720). A fix has just been checked into the trunk there.

**Comment by ilya-fedin** [comment]
The cherry-picks aren't merged yet, but the 6.5.0 one already failed, so it may be that it will be only in 6.5.1 (unless they will re-stage it tomorrow)

**Comment by ilya-fedin** [comment]
![изображение](https://user-images.githubusercontent.com/17829319/226859087-2c02a088-7a09-4fbd-b30c-7e1e7d5c4db3.png)
https://codereview.qt-project.org/c/qt/qtbase/+/468037

**Comment by ezaton** [comment]
I can add that the trigger on my computer is disconnecting and reconnecting the mouse several times. Scrolling does work on other applications correctly, however - not for Telegram.
I use KVM to connect multiple systems to the same keyboard/mouse, so after switching several times, this happens.
Maybe it will help in reproduction.

**Comment by ilya-fedin** [comment]
![изображение](https://user-images.githubusercontent.com/17829319/227181996-9cd3f5ab-04c8-4e63-97a4-3f5218a626d5.png)

**Comment by Aokromes** [comment]
fixed on 4.8.0

**Comment by nogurenn** [comment]
@Aokromes petition to reopen this issue? I think I found a lead.

The broken scrolling is reproducible using the Snap / sandboxed build (Ubuntu Desktop)

No issue when installing "directly"
```
tar -xf tsetup.4.8.1.tar.xz 
ls
sudo mv Telegram /opt/telegram
sudo ln -sf /opt/telegram/Telegram /usr/bin/telegram
```

**Comment by ilya-fedin** [comment]
@nogurenn this issue is not about closing when scrolling, it's about scrolling just not working. I.e. you scroll and nothing happens.
